CONTRIBUTING.md

```diff
@@ -55,6 +55,41 @@ docs(contributing-readme): add example of commit with subject line
 
 A commit hook makes sure the syntax is followed. Automated commit messages such as `Merge pull request` are handled.
 
+## Type generation
+This SDK support generating typescript types from the Cognite open api specification. The idea is to use openapi as a source of truth. Any incorrect or missing types should be fixed/added in the openapi spec instead of manually adjusting the generated types. This also helps to keep documentation up to date.
+
+The scripts prefix `codegen` are reserved for type generation. The core logic is found in `packages/codegen` with a deeper explenation.
+
+To quickly setup type generation for your service you may run the generate command with `--auto`:
+> Note: using `--auto` will setup a local versionfile and not the package global versionfile. You can later switch to the global versionfile once you have confirmed type generation works correctly for your service (you may also have to update your openapi spec).
+
+```sh
+yarn codegen --package=stable --version=v1 --service=documents --auto generate
+```
+
+A service must have a versionfile (essentially just a copy of the openapi spec) and a configuration (which openapi paths to parse). After
```

Comment:
I think it's a bit "hidden" in the text _which_ endpoints types are actually generated for. Can we make that more clear? Ideally I think we could shorten down all this text around.

Can a "service" start by only generating types for a single endpoint, or is it "all or nothing"?